### PR TITLE
[MOSIP-27961] Updated README.md to backup and restore kafka & softhsm

### PR DESCRIPTION
**Pull request by syedsalman3753**

### CODE REVIEW

File: deployment/v3/external/hsm/softhsm/README.md [hunk 1]
```diff
@@ -8,3 +8,30 @@ sh install.sh
 * Replication factor is 1.  Multiple replication factors are not supported on AWS at the moment 'cause AWS EBS does not support `ReadWriteMany`.
 * Keys are created in the mounted PV which gets mounted at `/softhsm/tokens` inside the container.
 * Random PIN generated if not specified. Set `securityPIN` in `values.yaml`.
+
+## Backup softhsm
```
Comment (SwatiPeriwal) [inline deployment/v3/external/hsm/softhsm/README.md:12]: ## Backup SoftHSM

File: deployment/v3/external/hsm/softhsm/README.md [hunk 2]
```diff
@@ -8,3 +8,30 @@ sh install.sh
...
+
+#### Backup 
+* Update below variable
```
Comment (SwatiPeriwal) [inline deployment/v3/external/hsm/softhsm/README.md:15]: Update the below variables

File: deployment/v3/external/hsm/softhsm/README.md [hunk 3]
```diff
@@ -8,3 +8,30 @@ sh install.sh
...
+  export POD_NAME=<pod-name>
+  ```
+* Execute the following command to create a backup of the SoftHSM to the cluster's console machine.
+  ```
+  kubectl --kubeconfig=$KUBECONFIG -n $NS cp $POD_NAME:softhsm/tokens ./softhsm-kernel/tokens
+  ```
+
+#### Restore
+* Ensure to have the backup directory for SoftHSM on cluster's console machine.
+* Update the variable provided below.
```
Comment (SwatiPeriwal) [inline deployment/v3/external/hsm/softhsm/README.md:28]: variables

File: deployment/v3/external/kafka/README.md [hunk 4]
```diff
@@ -9,3 +9,76 @@ Kafa is required only if you are installing Registration Processor.
 ```
 $ ./install.sh
 ```
+
+## Procedure to Backup & Restore bitnami kafka via Velero
```
Comment (SwatiPeriwal) [inline deployment/v3/external/kafka/README.md:13]: Procedure to Backup & Restore Bitnami Kafka via Velero

File: deployment/v3/external/kafka/README.md [hunk 5]
```diff
@@ -9,3 +9,76 @@ Kafa is required only if you are installing Registration Processor.
...
+
+#### Prerequisites
+* Required separate `minio` server / 'S3' , accessible from both backup & restore clusters along with `api` & `console` port accessible.
```
Comment (SwatiPeriwal) [inline deployment/v3/external/kafka/README.md:16]: Require separate `minio` server / `S3`, accessible from both backup & restore clusters along with accessible `api` & `console` port.

File: deployment/v3/external/kafka/README.md [hunk 6]
```diff
@@ -9,3 +9,76 @@ Kafa is required only if you are installing Registration Processor.
...
+* Ensure to create a bucket with name `velero`.
+* Required backup & restore servers cluster config file.
+* Install [velero](https://velero.io/docs/v1.9/basic-install/#install-the-cli) cli on both cluster management machines.
+* Install [minio client](https://docs.min.io/docs/minio-client-quickstart-guide.html) cli on both cluster management machines.
+
+#### Backup kafka
+* Login to the console machine,navigate to the `Kafka directory (under /v3/external/)`, and execute the backup script to create a backup of Kafka.
```
Comment (SwatiPeriwal) [inline deployment/v3/external/kafka/README.md:23]: Login to the console machine, navigate to..............

Space after comma needs to be added in the above sentence

Comment (SwatiPeriwal) [inline deployment/v3/external/kafka/README.md:23]: Add space after comma

File: deployment/v3/external/kafka/README.md [hunk 7]
```diff
@@ -9,3 +9,76 @@ Kafa is required only if you are installing Registration Processor.
...
+
+    [ Creating Backup ] 
+        Backup request "kafka-15-09-2022-13-21" submitted successfully.
+        Waiting for backup to complete. You may safely press ctrl-c to stop waiting - your backup will continue in the background.
+        .........................................
+        Backup completed with status: Completed. You may check for more information using the commands `velero backup describe kafka-15-09-2022-13-21` and `velero backup logs kafka-15-09-2022-13-21`.
+  ```
+
+
+#### Restore kafka
```
Comment (SwatiPeriwal) [inline deployment/v3/external/kafka/README.md:52]: Restore Kafka

**Review by SwatiPeriwal — CHANGES_REQUESTED** [review]
Minor changes

File: deployment/v3/external/object-store/minio/README.md [hunk 8]
```diff
@@ -83,3 +83,11 @@ The following environmental variables needs to be mapped for the same.
 ![minio-dashboard.png](images/minio-dashboard.png)
 
 The above mentioned changes are taken care in values.yaml script.
+
+
+## Troubleshoot
+* If you encounter the following error while attempting to access MinIO
+  ```
+  mc: <ERROR> Unable to list folder. The request signature we calculated does not match the signature you provided. Check your key and signing method
```
Comment (SwatiPeriwal) [inline deployment/v3/external/object-store/minio/README.md:91]: Add full stop

File: deployment/v3/external/object-store/minio/README.md [hunk 9]
```diff
@@ -83,3 +83,11 @@ The following environmental variables needs to be mapped for the same.
...
+  ```
+  Then, Ensure that you provide the appropriate API version, either `--api S3v2` or `--api S3v4`, when executing the `mc set alias` command.
```
Comment (SwatiPeriwal) [inline deployment/v3/external/object-store/minio/README.md:93]: No capitals for ensure here

File: deployment/v3/external/hsm/softhsm/README.md [hunk 10]
```diff
@@ -8,3 +8,30 @@ sh install.sh
...
 * Random PIN generated if not specified. Set `securityPIN` in `values.yaml`.
+
+## Backup SoftHSM
+
+#### Backup 
+* Update the below variables
+  ```
+  export KUBECONFIG=<kubeconfig-file>
+  export NS=<softhsm-namespace>
+  export POD_NAME=<pod-name>
```
Comment (ckm007) [inline deployment/v3/external/hsm/softhsm/README.md:19]: softhsm pod name

File: deployment/v3/external/hsm/softhsm/README.md [hunk 11]
```diff
@@ -8,3 +8,30 @@ sh install.sh
...
+  kubectl --kubeconfig=$KUBECONFIG -n $NS cp $POD_NAME:softhsm/tokens ./softhsm-kernel/tokens
+  ```
+
+#### Restore
+* Ensure to have the backup directory for SoftHSM on cluster's console machine.
+* Update the variables provided below.
+  ```
+  export KUBECONFIG=<kubeconfig-file>
+  export NS=<softhsm-namespace>
+  export POD_NAME=<pod-name>
```
Comment (ckm007) [inline deployment/v3/external/hsm/softhsm/README.md:32]: <destination softhsm pod name>

File: deployment/v3/external/hsm/softhsm/README.md (hunk 10 above)
Comment (ckm007) [reply deployment/v3/external/hsm/softhsm/README.md:19]: <softhsm source pod name>

Comment (ckm007) [reply deployment/v3/external/hsm/softhsm/README.md:19]: softhsm source pod name

File: deployment/v3/external/hsm/softhsm/README.md (hunk 11 above)
Comment (ckm007) [reply deployment/v3/external/hsm/softhsm/README.md:32]: softhsm destination pod name

**Comment by ckm007** [comment]
add section for removal of new keys from softhsm before colpying backed up keys.

**Pull request merged**